usage/rest-client/src/main/java/brooklyn/rest/client/BrooklynApi.java

```diff
@@ -116,22 +119,26 @@ public BrooklynApi(URL endpoint, ClientExecutor clientExecutor) {
         final T result0 = ProxyFactory.create(clazz, target, clientExecutor);
         return (T) Proxy.newProxyInstance(clazz.getClassLoader(), new Class<?>[] { clazz }, new InvocationHandler() {
             @Override
-            public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
-                Object result1;
+            public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {                 
                 try {
-                    result1 = method.invoke(result0, args);
+                    Object result1 = method.invoke(result0, args);
+                    Class<?> type = String.class;
+                    if (result1 instanceof Response) {
+                        if(((Response)result1).getStatus()/100 == 2) {
+                           String responseClass = method.getAnnotation(ApiOperation.class).responseClass();
```

Comment:
Looks like `responseClass` is used purely for documentation, can't rely on it for the exact return type. Better get the return type from the method handle.
For example:
  * `CatalogApi` doesn't include the fully qualified names
  * Some methods return collections, indicated by `multiValueResponse` in the annotation
  * Some have atomic values, not including a `responseClass`
  * Some methods have complex return types, but don't have `responseClass`
  * etc.